fastcache.go

```diff
@@ -117,12 +140,18 @@ type Cache struct {
 // since the cache holds data in memory.
 //
 // If maxBytes is less than 32MB, then the minimum cache capacity is 32MB.
-func New(maxBytes int) *Cache {
-	if maxBytes <= 0 {
-		panic(fmt.Errorf("maxBytes must be greater than 0; got %d", maxBytes))
+func New(config Config) *Cache {
```

Comment:
This a not backward compatible change.

Reply:
BTW, why not to try https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html ? 😉

Reply:
@cristaloleg thanks for the review. Yes, this PR introduces some change but maybe it's acceptable.  Since this an in-memory cache, so we don't have to consider too much compatible stuff. If "user" upgrades their package to the latest version, it's easy to use the new way to initialize the cache.

But appreciate for your link, will look it!

Reply:
Yeah, everything looks fine, but mention breaking changes is an important thing to do :)

Reply:
Yes sure. Actually, it's the first step to introduce the `Config`, then will add more config option if possible. 

Anyway thanks for reminding. I personally prefer the current approach now, to make the code as simple as possible.